File: server-java/src/main/java/io/maxi/examples/server/model/Team.java
package io.maxi.examples.server.model;

import io.maxi.annotation.MaxiField;
import io.maxi.annotation.MaxiType;

/**
 * Team DTO — mirrors the T:Team type in shared/sports.mxs.
 */
@MaxiType(alias = "T", name = "Team")
public class Team {

    @MaxiField(typeExpr = "int")
    public int id;

    @MaxiField(required = true)
    public String name;

    @MaxiField(required = true)
    public String city;

    @MaxiField(typeExpr = "int")
    public int founded;

    @MaxiField(required = true)
    public String coach;

    public Team() {}
}

File: server-php/src/Model/Team.php
<?php

declare(strict_types=1);

namespace App\Model;

use Maxi\Attribute\MaxiField;
use Maxi\Attribute\MaxiType;

/**
 * Team DTO — mirrors the T:Team type in shared/sports.mxs.
 */
#[MaxiType(alias: 'T', name: 'Team')]
class Team
{
    public function __construct(
        #[MaxiField(typeExpr: 'int')]
        public int $id = 0,

        #[MaxiField(required: true)]
        public string $name = '',

        #[MaxiField(required: true)]
        public string $city = '',

        #[MaxiField(typeExpr: 'int')]
        public int $founded = 0,

        #[MaxiField(required: true)]
        public string $coach = '',
    ) {}
}
